File: MoEPluginGeofence/MoEGeofenceBridge.m
//
//  MoEGeofenceBridge.m
//  MoEPluginGeofence
//
//  Created by Rakshitha on 15/12/21.
//

#import <Foundation/Foundation.h>
#import "MoEGeofenceBridge.h"
#import <MoEPluginBase/MoEPluginBase.h>

@implementation MoEGeofenceBridge

+(instancetype)sharedInstance{
    static dispatch_once_t onceToken;
    static MoEGeofenceBridge* instance;
    dispatch_once(&onceToken, ^{
        instance = [[MoEGeofenceBridge alloc] init];
    });
    return instance;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
    }
    return self;
}

- (void)startGeofenceMonitoring:(NSDictionary*)geofenceDict {
    NSString* appID = [MoEPluginUtils getAppID: geofenceDict];
    if (appID == nil || appID.length <= 0) {
        [MOLogger debug:@"AppID is not available" label:kLoggerPluginBase sdkConfig:nil];
        return;
    }
    [[MOGeofence sharedInstance] startGeofenceMonitoringForAppID:appID];
}
@end

File: MoEPluginGeofence/MoEGeofenceBridge.swift
//
//  MoEGeofenceBridge.swift
//  MoEngageGeofencePlugin
//
//  Created by Rakshitha on 23/06/22.
//

import Foundation
import MoEngageGeofence
import MoEPluginBase

@objc final public class MoEGeofenceBridge: NSObject, MoEPluginUtils {
    
    @objc public static let sharedInstance = MoEGeofenceBridge()
    
    private override init() {
        super.init()
    }
    
    @objc public func startGeofenceMonitoring(_ payload: [String: Any]) {
        let identifier = MoEGeofenceBridge.fetchIdentifier(attribute: payload)
        MOGeofence.sharedInstance.startGeofenceMonitoring(forAppID: identifier)
    }
}
